src/bots/userPnlSettler.ts

```diff
@@ -183,317 +248,185 @@ export class UserPnlSettlerBot implements Bot {
-					let pnlToSettleWithUser = ZERO;
-					let pnlPoolTookenAmount = ZERO;
-					let settleUser = false;
-					if (userUnsettledPnl.gt(ZERO)) {
-						pnlPoolTookenAmount = getTokenAmount(
-							pnlPool.scaledBalance,
-							spotMarketAndOracleData[pnlPool.marketIndex].marketAccount,
-							SpotBalanceType.DEPOSIT
-						);
-						pnlToSettleWithUser = BN.min(userUnsettledPnl, pnlPoolTookenAmount);
-						if (pnlToSettleWithUser.gt(ZERO)) {
-							const netUserPnl = calculateNetUserPnl(
-								perpMarketAndOracleData[perpMarketIdx].marketAccount,
-								perpMarketAndOracleData[perpMarketIdx].oraclePriceData
-							);
-							let maxPnlPoolExcess = ZERO;
-							if (netUserPnl.lt(pnlPoolTookenAmount)) {
-								maxPnlPoolExcess = pnlPoolTookenAmount.sub(
-									BN.max(netUserPnl, ZERO)
-								);
-							}
-
-							// we're only allowed to settle positive pnl if pnl pool is in excess
-							if (maxPnlPoolExcess.lte(ZERO)) {
-								logger.warn(
-									`Want to settle positive PnL for user ${user
-										.getUserAccountPublicKey()
-										.toBase58()} in market ${perpMarketIdx}, but maxPnlPoolExcess is: (${convertToNumber(
-										maxPnlPoolExcess,
-										QUOTE_PRECISION
-									)})`
-								);
-								continue;
-							}
-
-							const netUserPnlImbalance = calculateNetUserPnlImbalance(
-								perpMarketAndOracleData[perpMarketIdx].marketAccount,
-								spotMarketAndOracleData[spotMarketIdx].marketAccount,
-								perpMarketAndOracleData[perpMarketIdx].oraclePriceData
-							);
-							if (netUserPnlImbalance.gt(ZERO)) {
-								logger.warn(
-									`Want to settle positive PnL for user ${user
-										.getUserAccountPublicKey()
-										.toBase58()} in market ${perpMarketIdx}, protocol's AMM lacks excess PnL (${convertToNumber(
-										netUserPnlImbalance,
-										QUOTE_PRECISION
-									)})`
-								);
-								continue;
-							}
-						}
-
-						// only settle user pnl if they have enough collateral
-						if (user.canBeLiquidated().canBeLiquidated) {
-							logger.warn(
-								`Want to settle negative PnL for user ${user
-									.getUserAccountPublicKey()
-									.toBase58()}, but they have insufficient collateral`
-							);
-							continue;
-						}
-						settleUser = true;
-					} else {
-						// only settle negative pnl if unsettled pnl is material
-						if (userUnsettledPnl.abs().gte(this.minPnlToSettle.abs())) {
-							logger.info(
-								`Settling negative pnl for user ${user
-									.getUserAccountPublicKey()
-									.toBase58()} in market ${
-									settleePosition.marketIndex
-								} with unsettled pnl: ${convertToNumber(
-									userUnsettledPnl,
-									QUOTE_PRECISION
-								)}`
-							);
-							settleUser = true;
-						}
-					}
+				// Only settle if user's free margin is less than 10% of account value
+				if (
+					userFreeMarginCurrent >=
+					userAccountValue * MIN_MARGIN_RATIO_FOR_POSITIVE_PNL
+				) {
+					continue;
```

Comment:
i think you also want to check that `positiveOnly == true` when doing lowmargin check? otherwise you're skipping all the users for neg pnl too